src/Microsoft.AspNetCore.Mvc.Analyzers/CodeAnalysisExtensions.cs

```diff
@@ -123,6 +123,24 @@ public static bool IsAssignableFrom(this ITypeSymbol source, ITypeSymbol target)
             return false;
         }
 
+        public static bool IsCollection(this ITypeSymbol source)
```

Comment:
`IsEnumerable` would be a better name. May also need `IsCollection` and `IsArray` to distinguish the cases where current `MVC1005_ParameterAttributeAvoidUsingRequiredOncollections` advice is actionable under .NET Core and .NET Framework (respectively).

Reply:
Could we move this method to the analyzer? The methods on this type are meant to be more general purpose.